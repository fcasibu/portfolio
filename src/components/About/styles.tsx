import styled, { keyframes } from 'styled-components';

const slideIn = keyframes`
  from {
    transform: translateX(500px);
  }

  to {
    transform; translateX(0);
  }
`;

const fadeIn = keyframes`
  from {
    transform: translateY(40px);
  }

  to {
    opacity: 1;
    transform: translateY(0);
  }
`;

export const StyledAbout = styled.div`
  background: var(--theme-background);
  display: flex;
  flex-direction: column;
  gap: 1rem;
  padding: 1rem 1.5rem;
  width: 50vw;
  animation: ${slideIn} 0.5s ease-in-out forwards;

  & > * {
    opacity: 0;
    animation: ${fadeIn} 1.2s ease-in-out forwards 0.3s;
  }

  @media (max-width: 700px) {
    width: 100%;
    height: 90vh;
  }
`;

export const StyledAboutInfo = styled.div``;

export const StyledLanguagesAndTools = styled.div`
  display: flex;
  flex-direction: column;
`;

export const AvatarWrapper = styled.div`
  display: flex;
  align-items: center;

  @media (max-width: 700px) {
    flex-direction: column;
  }
`;

export const Avatar = styled.img`
  max-width: 125px;
  max-height: 125px;
`;
