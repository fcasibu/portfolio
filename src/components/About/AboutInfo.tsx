import { Para } from '../Para';
import { Avatar, AvatarWrapper, StyledAboutInfo } from './styles';

interface Props {
  info1: string;
  info2: string;
  avatar: string;
}

export const AboutInfo = ({ info1, info2, avatar }: Props) => {
  return (
    <StyledAboutInfo>
      <AvatarWrapper>
        <Avatar src={avatar} alt="" />
        <Para>{info1}</Para>
      </AvatarWrapper>
      <Para>{info2}</Para>
    </StyledAboutInfo>
  );
};
