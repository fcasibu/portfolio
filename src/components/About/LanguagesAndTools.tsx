import Icons from '../Icons';
import { Title } from '../Title';
import { StyledLanguagesAndTools } from './styles';

interface Props {
  icons: NodeRequire[];
  title: string;
}

export const LanguagesAndTools = ({ icons, title }: Props) => {
  return (
    <StyledLanguagesAndTools>
      <Title>{title}</Title>
      <Icons icons={icons} size="medium" state="block" />
    </StyledLanguagesAndTools>
  );
};
