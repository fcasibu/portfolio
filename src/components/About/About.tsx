import { AboutInfo } from './AboutInfo';
import { LanguagesAndTools } from './LanguagesAndTools';
import { StyledAbout } from './styles';
import { aboutData } from '../../data/data';
import { Title } from '../Title';

const About = () => {
  return (
    <StyledAbout>
      <Title>{aboutData.titles.about}</Title>
      <AboutInfo info1={aboutData.info1} info2={aboutData.info2} avatar={aboutData.avatar} />
      <LanguagesAndTools icons={aboutData.ltIcons} title={aboutData.titles.lt} />
    </StyledAbout>
  );
};

export default About;
