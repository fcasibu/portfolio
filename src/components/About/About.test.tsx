import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import About from './About';
import { AboutInfo } from './AboutInfo';

describe('About Me', () => {
  it('should render properly and match the snapshot', () => {
    const { container } = render(<About />);

    expect(container).toMatchSnapshot();
  });

  it('should render the info correctly', () => {
    render(<AboutInfo info1="Hello" info2="World" avatar="" />);

    const info1 = screen.getByText(/hello/i);
    const info2 = screen.getByText(/world/i);

    expect(info1).toBeInTheDocument();
    expect(info2).toBeInTheDocument();
  });
});
