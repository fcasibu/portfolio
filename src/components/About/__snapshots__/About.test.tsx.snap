// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Me should render properly and match the snapshot 1`] = `
<div>
  <div
    class="sc-gsnTZi jhBbNc"
  >
    <h2
      class="sc-papXJ ldZqnn"
    >
      About me
    </h2>
    <div
      class="sc-dkzDqf"
    >
      <div
        class="sc-eCYdqJ dilfOq"
      >
        <img
          alt=""
          class="sc-jSMfEi bfPLwU"
          src="avatar-default-icon.png"
        />
        <p
          class="sc-bczRLJ iGcDty"
        >
          Hello there! I am Francis Eugene Casibu, a full stack web developer based in the Philippines. An undergrad studying Information Technology. I have a passion for developing web applications, and fully committed to lifelong learning.
        </p>
      </div>
      <p
        class="sc-bczRLJ iGcDty"
      >
        I started my Web Development journey on January 8, 2022 and have been going strong ever since. When I am off coding, I enjoy reading light novels, working out, playing basketball, and helping other learners achieve their goals in The Odin Project discord.
      </p>
    </div>
    <div
      class="sc-hKMtZM cfGbOi"
    >
      <h2
        class="sc-papXJ ldZqnn"
      >
        Languages and Tools
      </h2>
      <ul
        class="sc-gKXOVf eYidII"
      >
        <li
          class="sc-iBkjds"
        >
          <img
            alt=""
            class="sc-ftvSup ePeZbz"
            src="html.svg"
          />
        </li>
        <li
          class="sc-iBkjds"
        >
          <img
            alt=""
            class="sc-ftvSup ePeZbz"
            src="css.svg"
          />
        </li>
        <li
          class="sc-iBkjds"
        >
          <img
            alt=""
            class="sc-ftvSup ePeZbz"
            src="javascript.svg"
          />
        </li>
        <li
          class="sc-iBkjds"
        >
          <img
            alt=""
            class="sc-ftvSup ePeZbz"
            src="typescript.svg"
          />
        </li>
        <li
          class="sc-iBkjds"
        >
          <img
            alt=""
            class="sc-ftvSup ePeZbz"
            src="react.svg"
          />
        </li>
        <li
          class="sc-iBkjds"
        >
          <img
            alt=""
            class="sc-ftvSup ePeZbz"
            src="node.svg"
          />
        </li>
        <li
          class="sc-iBkjds"
        >
          <img
            alt=""
            class="sc-ftvSup ePeZbz"
            src="mongodb.svg"
          />
        </li>
        <li
          class="sc-iBkjds"
        >
          <img
            alt=""
            class="sc-ftvSup ePeZbz"
            src="jest.svg"
          />
        </li>
        <li
          class="sc-iBkjds"
        >
          <img
            alt=""
            class="sc-ftvSup ePeZbz"
            src="firebase.svg"
          />
        </li>
        <li
          class="sc-iBkjds"
        >
          <img
            alt=""
            class="sc-ftvSup ePeZbz"
            src="git.svg"
          />
        </li>
        <li
          class="sc-iBkjds"
        >
          <img
            alt=""
            class="sc-ftvSup ePeZbz"
            src="webpack.svg"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
